core-services/src/main/java/org/zalando/nakadi/service/publishing/EventPublisher.java

```diff
@@ -132,8 +128,8 @@ EventPublishResult processInternal(final String events,
             }
             validateEventOwnership(eventType, batch);
             validate(batch, eventType, delete);
+            assignKeys(batch, eventType);
```

Comment:
Changed the order, so that we can later calculate partitions based on the event keys.